### ISSUE TITLE
Blender 2.8 RC2 - struct.error: unpack requires a buffer of 16 bytes

### ISSUE THREAD

**Issue by 0Curtain0** [xpawelsky/RipImport#1]
**Blender v2.80 Release Candidate 2**
**RipImport v0.2.1** (Jan 20th 2019)
**Ubuntu Linux 64bit**

Hey, I was trying to test the add-on with the recent Blender RC, and i got the following error when attempting to import a .rip file:
```
Traceback (most recent call last):
  File "/home/user/.config/blender/2.80/scripts/addons/io_import_rip/__init__.py", line 101, in execute
    return import_rip.load(self, context, **keywords)
  File "/home/user/.config/blender/2.80/scripts/addons/io_import_rip/import_rip.py", line 288, in load
    context,
  File "/home/user/.config/blender/2.80/scripts/addons/io_import_rip/import_rip.py", line 240, in load_rip
    read_rip_file(file, filepath_basename[:-4], texture_search_path)
  File "/home/user/.config/blender/2.80/scripts/addons/io_import_rip/import_rip.py", line 52, in read_rip_file
    signature, version = struct.unpack('LL', file.read(8))
struct.error: unpack requires a buffer of 16 bytes

location: <unknown location>:-1
```

**Comment by 0Curtain0** [comment]
There is another user having a similar error on a Mac computer:
```
Traceback (most recent call last):
 File "/Users/user/Library/Application Support/Blender/2.80/scripts/addons/io_import_rip/__init__.py", line 101, in execute
   return import_rip.load(self, context, **keywords)
 File "/Users/user/Library/Application Support/Blender/2.80/scripts/addons/io_import_rip/import_rip.py", line 285, in load
   load_rip(os.path.join(ripdir, file), context)
 File "/Users/user/Library/Application Support/Blender/2.80/scripts/addons/io_import_rip/import_rip.py", line 240, in load_rip
   read_rip_file(file, filepath_basename[:-4], texture_search_path)
 File "/Users/user/Library/Application Support/Blender/2.80/scripts/addons/io_import_rip/import_rip.py", line 52, in read_rip_file
   signature, version = struct.unpack('LL', file.read(8))
struct.error: unpack requires a buffer of 16 bytes

location: <unknown location>:-1
```

**Comment by luchador3d** [comment]
yes, I had this problem too.  if you double all the file.read(x) values it will eventually break on other things.  It's like it's using the wrong version of python or something.

**Comment by xpawelsky** [comment]
I can't reproduce the issue. Plugin is still working without any changes while using Blender 2.8 official release and Windows 10 1093

**Comment by luchador3d** [comment]
I’m on a Mac if that makes any difference.  Is there anything you’d like me
to check?

**Comment by 0Curtain0** [comment]
**Blender v2.80 Official Release**
**RipImport v0.2.1** https://github.com/xpawelsky/RipImport/releases/download/0.2.1/io_import_rip-2-8.zip 
**Pop!_OS 64bit** (Ubuntu Linux) 

```
Traceback (most recent call last):
  File "/home/user/.config/blender/2.80/scripts/addons/io_import_rip/__init__.py", line 101, in execute
    return import_rip.load(self, context, **keywords)
  File "/home/user/.config/blender/2.80/scripts/addons/io_import_rip/import_rip.py", line 288, in load
    context,
  File "/home/user/.config/blender/2.80/scripts/addons/io_import_rip/import_rip.py", line 240, in load_rip
    read_rip_file(file, filepath_basename[:-4], texture_search_path)
  File "/home/user/.config/blender/2.80/scripts/addons/io_import_rip/import_rip.py", line 52, in read_rip_file
    signature, version = struct.unpack('LL', file.read(8))
struct.error: unpack requires a buffer of 16 bytes

location: <unknown location>:-1
```
Something else I noticed, in **Blender Preferences** - **Add-ons**, the RipImport add-on shows version **0.0.2**. 

I'm not a programmer, but like luchador3d, is there anything you would like me to check? 
Maybe some new dependency I need to install? 
The add-on used to work during the Blender 2.80 beta.

**Comment by Dummiesman** [comment]
When I used Linux for a bit I realized that this is due to the way struct itself works, I had to replace things like LL with <LL for proper endianness and packing.

**Comment by 0Curtain0** [comment]
@Dummiesman 
Hey, good to see you're alive, and thank you for commenting. I don't know the technicalities of what has been said, but is it possible to commit a fix?

**Comment by Dummiesman** [comment]
Unfortunately I don't have an active Linux environment to test on so I wouldn't be able to do much at the moment. At some point I may make a VM because this also affects a couple of my plugins.

**Comment by luchador3d** [comment]
I got it a little further but have gotten stuck.  hopefully this helps with further debugging

In general, on 64bit osx you'll need to add = to the begining of the struct statements
```face_count, vertex_count, vertex_size, textures_count, shaders_count, attributes_count = struct.unpack('=LLLLLL', file.read(24))```

Second the read_string function is misbehaving so i simplified it to this for debugging


```
def read_string(file):

	read_str = struct.unpack("8s", file.read(8) )[0].decode("ascii")
	file.seek(32)
	return read_str
```
this takes the next 32 bytes from the file io as C characters, breaks them out of the tuple, and casts the result to an ascii string which is what python is happy comparing later.  It also moves the file pointer back to position 32 so the rest of the offset line up again.

if you do this it can successfully find TEXCOORD from the rip files (for example) and proceed with the UV layout.  however it looks like it's doing that layout wrong because no tris show up in blender.  whatever the case, the method you had previously looked like it was grabbing twice as much data as it should on osx which broke everything.

basically i think it all boils down to ripping things on a 32bit env and then parsing them on a 64 bit env is going to suck just a little.  it's probably something only mac guys have to worry about because they're using a vm to generate the rips and then pulling them into the native 64bit env for parsing.

**Comment by luchador3d** [comment]
Attached example rip files generated on 32bit system which will not parse using this on blender 2.8 in 64 bit system.
[Archive.zip](https://github.com/xpawelsky/RipImport/files/3828762/Archive.zip)
